include/swift/Sema/TypeCheckRequest.h

```diff
@@ -97,6 +98,9 @@ class TypeCheckRequest {
 
 #include "swift/Sema/TypeCheckRequestKinds.def"
     }
+
+    // Work around MSVC warning: not all control paths return a value
+    llvm_unreachable("All switch cases are covered");
```

Comment:
Another nit: I understand where you're coming from with the particular text here (and in the other places you added llvm_unreachable) since it does appear to be currently covered, but generally we tend to word these as something like "Unhandled <kind,type,etc.> in switch!" so that if the unreachable is ever hit the message is clear.